monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/TypeFilter.tsx

```diff
@@ -0,0 +1,65 @@
+import React, {useEffect, useState} from 'react';
+import SelectComponent from '../Select';
+import _ from 'lodash';
+
+type TypeFilterProps = {
+  allPlugins: any[],
+  displayedPlugins: any[],
+  setDisplayedPlugins: (plugins: any[]) => void,
+  className?: string
+}
+
+type SelectOption = {
+  value: string,
+  label: string
+}
+
+const anyTypeOption :SelectOption = {value: "", label: "All"}
+
+const TypeFilter = (props: TypeFilterProps) => {
+  const [selectedType, setSelectedType] = useState(anyTypeOption)
+  const [typeFilters, setTypeFilters] = useState([])
+
+  useEffect(() => {
+    let allTypes = [];
+    allTypes = props.allPlugins.map(plugin => plugin.type_ || plugin.plugin_type)
```

Comment:
Use the `extractPluginsPropertyValues` function.
It will return a sorted array with unique values.

Reply:
From `mocksHelper.js`?  It doesn't support multiple properties. This already returns a sorted array